### ISSUE TITLE
[Frameworks] Error building framework from clean state, succeeds on next build

### ISSUE THREAD

**Issue by mowens** [CocoaPods/CocoaPods#3035]
Error when building framework with pods from a clean state but succeeds to build by rerunning the build (without cleaning). The build from a clean state is unable to find the pod framework:

![screen shot 2015-01-13 at 2 07 50 pm](https://cloud.githubusercontent.com/assets/4575216/5730336/1f1b95d0-9b2e-11e4-96d8-12babba82756.png)

Reproducible in the following situation:
1. PodA depends on Objective-C framework
2. PodB depends on PodA, Objective-C framework and Swift frameworks

I have created demo projects for both PodA and PodB to illustrate the problem:
https://github.com/mowens/PodA
https://github.com/mowens/PodB

Both projects have the generated Pods directory checked in as well for debugging purposes.

To reproduce:
1. Checkout PodB project
2. Open workspace
3. Clean build
4. Build

Expected:
Builds without failures

Actual:
Build fails unable to find the Typhoon framework (Objective-C framework dependency in both PodA and PodB)

Attempting a rebuild will build PodB successfully.

**Comment by fatuhoku** [comment]
To add to this issue, I found that external frameworks don't get linked if they're a dependency of subspecs.

So e.g. if `Pod A` has a subspec that depends on the `Parse` framework...
```
# PodA.podspec

...
  s.subspec 'ParseComponents' do |ss|
    ...
    ss.dependency 'Parse'
  end
...
```

and `Pod B` depends on `Pod A`, then it doesn't matter how many times you try to build `Pod B` (I've tried completely clean builds too), you will get linker errors for Parse classes (the trace looks like https://github.com/CocoaPods/Specs/issues/1857).

**Comment by mowens** [comment]
I invested the issue a bit more and tested this against the latest release 0.36.0 and I found that the frameworks are building correctly, however, there is a linker issue with how CocoaPods sets up the links.

CocoaPods will define the pod framework build path as so:

```
PODS_FRAMEWORK_BUILD_PATH = $(BUILD_DIR)/$(CONFIGURATION)$(EFFECTIVE_PLATFORM_NAME)/Pods
```

This path is added to the ```FRAMEWORK_SEARCH_PATHS```. The dependency Pod frameworks do in fact get built *and* exist in the ```PODS_FRAMEWORK_BUILD_PATH``` directory, however, when compiling the Swift source of the parent project (in this case a Swift dynamic framework itself) it is unable to find any of the frameworks that exist in the ```PODS_FRAMEWORK_BUILD_PATH```.

Xcode *is* able to find frameworks and link against them properly *if* the frameworks are built to the ```BUILT_PRODUCTS_DIR```. These are what the build settings look like:

```
BUILT_PRODUCTS_DIR = /Users/mikeowens/Library/Developer/Xcode/DerivedData/Foo-eokprkgtddxobafmkyrhziicbjbr/Build/Products/Debug-iphonesimulator
FRAMEWORK_SEARCH_PATHS = "/Users/mikeowens/Library/Developer/Xcode/DerivedData/Foo-eokprkgtddxobafmkyrhziicbjbr/Build/Products/Debug-iphonesimulator/Pods"
PODS_FRAMEWORK_BUILD_PATH = /Users/mikeowens/Library/Developer/Xcode/DerivedData/Foo-eokprkgtddxobafmkyrhziicbjbr/Build/Products/Debug-iphonesimulator/Pods
```

Essentially the Pods directory is a child directory of the ```BUILD_PRODUCTS_DIR``` and this seems to be causing the linking issue. 

As a very, very, very hacky work around to unblock my builds, I am manually copying the build pods frameworks out of the Pods directory and into the parent directory before the parent project compiles its Swift source.

```
CURRENT_DIR=`pwd`
cd $PODS_FRAMEWORK_BUILD_PATH
cp -rf * ..
cd $CURRENT_DIR
```

**Comment by neonichu** [comment]
This looks like a duplicate of #3095 to me.

**Comment by mowens** [comment]
Seems like it could be related. The errors are slightly different:

![image](https://cloud.githubusercontent.com/assets/4575216/6609417/5aa7e876-c80e-11e4-92be-e2e85b2b669c.png)

**Comment by Crufty** [comment]
I'm seeing a similar problem trying to build a framework that uses some CocoaPods (also built as frameworks, using v0.36). When it builds, the frameworks end up in:

    $(BUILD_DIR)/$(CONFIGURATION)$(EFFECTIVE_PLATFORM_NAME)
    e.g. ~/Library/Developer/Xcode/DerivedData/<gobldeygook>/Build/Products/Foo.framework

and a weird thing is happening where it creates a partial framework folder containing just a `Modules` folder and nothing else (no headers or binaries) in:

    $(BUILD_DIR)/$(CONFIGURATION)$(EFFECTIVE_PLATFORM_NAME)/Pods
    e.g. ~/Library/Developer/Xcode/DerivedData/<gobldeygook>/Build/Products/Pods/Foo.framework

This second path is the one passed in `FRAMEWORK_SEARCH_PATHS` when compiling which then fails as it cannot find the headers.

I have been unable to pinpoint why the completed frameworks are in the main build folder, and these partial frameworks are appearing under the `Pods` subfolder. It seems to me that part of the toolchain is using the `PODS_FRAMEWORK_BUILD_PATH` and part of it is using the default build output folder instead. Or perhaps one of my targets other project in the workspace is misconfigured and causing this.

As a complete hack/satisfactory workaround I edited the `Pods.*.xcconfig` files and each individual pod's `Pods-Foo-Private.xcconfig` files and deleted the `/Pods` suffix from `PODS_FRAMEWORK_BUILD_PATH`

This is working for me now after doing this.

**Comment by neonichu** [comment]
It is expected behaviour for frameworks build using CocoaPods to end up in a subdirectory, see linked issue #3095

**Comment by segiddins** [comment]
Is this still an issue?
